src/pudl/transform/ferc714.py

```diff
@@ -315,35 +361,223 @@ def _post_process(df: pd.DataFrame, table_name: str) -> pd.DataFrame:
     return PUDL_PACKAGE.get_resource(table_name).enforce_schema(df)
 
 
+class RespondentId:
+    """Class for building the :ref:`out_ferc714__hourly_planning_area_demand` asset.
+
+    Most of the methods in this class as staticmethods. The purpose of using a class
+    in this instance is mostly for organizing the table specific transforms under the
+    same name-space.
+    """
+
+    @classmethod
+    def run(
+        cls, raw_csv: pd.DataFrame, raw_xbrl_duration: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Build the table for the :ref:`core_ferc714__respondent_id` asset.
+
+        Process and combine the CSV and XBRL based data.
+
+        There are two main threads of transforms happening here:
+
+        * Table compatibility: The CSV raw table is static (does not even report years)
+          while the xbrl table is reported annually. A lot of the downstream analysis
+          expects this table to be static. So the first step was to check whether or not
+          the columns that we have in the CSV years had consistent data over the few XBRL
+          years that we have. There are a small number of eia_code's we needed to clean
+          up, but besides that it was static. We then convert the XBRL data into a static
+          table, then we concat-ed the tables and checked the static-ness again via
+          :meth:`ensure_eia_code_uniqueness`.
+        * eia_code cleaning: Clean up FERC-714 respondent names and manually assign EIA
+          utility IDs to a few FERC Form 714 respondents that report planning area demand,
+          but which don't have their corresponding EIA utility IDs provided by FERC for
+          some reason (including PacifiCorp). Done all via :meth:`spot_fix_eia_codes` &
+          EIA_CODE_FIXES.
+
+        """
+        table_name = "core_ferc714__respondent_id"
+        # CSV STUFF
+        csv = (
+            _pre_process_csv(raw_csv, table_name)
+            .pipe(_assign_respondent_id_ferc714, source="csv")
+            .astype({"eia_code": pd.Int64Dtype()})
+            .pipe(cls.spot_fix_eia_codes, "csv")
+            .pipe(cls.ensure_eia_code_uniqueness, "csv")
+            .assign(source="csv")
+        )
+        # XBRL STUFF
+        xbrl = (
+            rename_columns(
+                raw_xbrl_duration,
+                params=RenameColumns(columns=RENAME_COLS[table_name]["xbrl"]),
+            )
+            .pipe(_assign_respondent_id_ferc714, source="xbrl")
+            .pipe(cls.clean_eia_codes_xbrl)
+            .astype({"eia_code": pd.Int64Dtype()})
+            .pipe(cls.spot_fix_eia_codes, "xbrl")
+            .pipe(cls.ensure_eia_code_uniqueness, "xbrl")
+            .pipe(cls.convert_into_static_table_xbrl)
+            .assign(source="xbrl")
+        )
+        # CONCATED STUFF
+        df = (
+            pd.concat([csv, xbrl])
+            .reset_index(drop=True)
+            .convert_dtypes()
+            .pipe(cls.spot_fix_eia_codes, "combined")
```

Comment:
What's the rationale for having a "combined" spot fix if you are also going to have breakout xbrl and csv spot fixes? Would be nice to add a little sentence about this in the `EIA_CODE_FIXES` dict above.

Reply:
i thought about combining them all of keeping them all separate. I liked this option the best because i could only add things to the combined section if they actually needed to get fixed in both sources. I will add docs about it!